server/model/todoModel.ts

```diff
@@ -0,0 +1,37 @@
+import { Types, Schema, model, Model } from 'mongoose';
+
+enum TodoPriority {
+  High = 'High',
+  Medium = 'Medium',
+  Low = 'Low',
+}
+
+interface ITodo extends Document {
+  title: string;
+  description: string;
+  priority: string;
+  date: string;
+  isChecked: boolean;
+  user: Types.ObjectId;
+}
+
+const todoSchema: Schema = new Schema({
+  title: { type: String, default: 'Title', required: true },
+  description: { type: String, default: 'Description', required: true },
+  priority: {
+    type: String,
+    required: true,
+    enum: Object.values(TodoPriority),
+    default: TodoPriority.Medium,
+  },
+  date: {
```

Comment:
if a property is required you don't need to have a default value for it. The field value will be the default if you don't provide a value, but if it's required then you always need to provide a value.

Reply:
also change the type of date to Date please